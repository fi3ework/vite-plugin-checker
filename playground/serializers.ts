import os from 'os'

const winNewLineReg = /\/r\/n/gim
const winSepReg = /\\/g

function doesUseDoubleSlashAsPath(val: string) {
  return val.includes('//vite-plugin-checker//')
}

export const normalizeWindowsLogSerializer = {
  print(val: string, serialize) {
    let result = val
    if (os.platform() === 'win32') {
      result = result.replaceAll(winNewLineReg, '/n')
      result = result.replaceAll(process.cwd().replace(winSepReg, '/'), '<PROJECT_ROOT>')

      if (doesUseDoubleSlashAsPath(result)) {
        result = result.replaceAll(
          `//a//vite-plugin-checker//vite-plugin-checker//playground-temp`,
          '<PROJECT_ROOT>/playground-temp'
        )
        result = result.split('//').join('/')
      }
    }

    return serialize(result)
  },
  test(val) {
    if (typeof val !== 'string') return false

    if (
      (os.platform() === 'win32' &&
        (val.includes(process.cwd().replace(winSepReg, '/')) || winNewLineReg.test(val))) ||
      doesUseDoubleSlashAsPath(val)
    ) {
      return true
    }

    return false
  },
}
