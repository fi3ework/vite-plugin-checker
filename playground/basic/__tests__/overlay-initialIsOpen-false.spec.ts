/* eslint-disable max-nested-callbacks */
import {
  getHmrOverlayText,
  killServer,
  preTest,
  viteServe,
} from 'vite-plugin-checker/__tests__/e2e/Sandbox/Sandbox'
import {
  editFile,
  sleep,
  testDir,
  WORKER_CLEAN_TIMEOUT,
} from 'vite-plugin-checker/__tests__/e2e/testUtils'

import { copyCode } from '../../../scripts/jestSetupFilesAfterEnv'
import { serializers } from '../../../scripts/serializers'

beforeAll(async () => {
  await preTest()
})

expect.addSnapshotSerializer(serializers)

afterAll(async () => {
  await sleep(WORKER_CLEAN_TIMEOUT)
})

describe('overlay-initialIsOpen-false', () => {
  beforeEach(async () => {
    await copyCode()
  })

  describe('serve', () => {
    afterEach(async () => {
      await killServer()
    })

    it('get initial error and subsequent error', async () => {
      editFile('vite.config.ts', (code) =>
        code.replace(`// edit-slot`, `overlay: { initialIsOpen: false },`)
      )
      await viteServe({
        cwd: testDir,
        launchPage: true,
      })
      await sleep(6000)
      const [message, file, frame] = await getHmrOverlayText()
      expect(message).toMatchSnapshot()
      expect(file).toMatchSnapshot()
      expect(frame).toMatchSnapshot()
    })
  })
})
