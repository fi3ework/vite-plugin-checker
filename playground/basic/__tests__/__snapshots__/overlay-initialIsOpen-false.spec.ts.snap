// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overlay-initialIsOpen-false serve get initial error and subsequent error 1`] = `""`;

exports[`overlay-initialIsOpen-false serve get initial error and subsequent error 2`] = `""`;

exports[`overlay-initialIsOpen-false serve get initial error and subsequent error 3`] = `""`;
